src/main/scala/akka/persistence/journal/JournalSpec.scala

```diff
@@ -38,97 +39,103 @@ trait JournalSpec extends PluginSpec {
       1 to 3 foreach { i => receiverProbe.expectMsg(replayedMessage(i)) }
-      receiverProbe.expectMsg(ReplayMessagesSuccess)
+      receiverProbe.expectMsg(RecoverySuccess(5))
     }
     "replay messages using a count limit" in {
       journal ! ReplayMessages(1, Long.MaxValue, 3, pid, receiverProbe.ref)
       1 to 3 foreach { i => receiverProbe.expectMsg(replayedMessage(i)) }
-      receiverProbe.expectMsg(ReplayMessagesSuccess)
+      receiverProbe.expectMsg(RecoverySuccess(5))
     }
     "replay messages using a lower and upper sequence number bound" in {
       journal ! ReplayMessages(2, 4, Long.MaxValue, pid, receiverProbe.ref)
       2 to 4 foreach { i => receiverProbe.expectMsg(replayedMessage(i)) }
-      receiverProbe.expectMsg(ReplayMessagesSuccess)
+      receiverProbe.expectMsg(RecoverySuccess(5))
     }
     "replay messages using a lower and upper sequence number bound and a count limit" in {
       journal ! ReplayMessages(2, 4, 2, pid, receiverProbe.ref)
       2 to 3 foreach { i => receiverProbe.expectMsg(replayedMessage(i)) }
-      receiverProbe.expectMsg(ReplayMessagesSuccess)
+      receiverProbe.expectMsg(RecoverySuccess(5))
     }
     "replay a single if lower sequence number bound equals upper sequence number bound" in {
       journal ! ReplayMessages(2, 2, Long.MaxValue, pid, receiverProbe.ref)
       2 to 2 foreach { i => receiverProbe.expectMsg(replayedMessage(i)) }
-      receiverProbe.expectMsg(ReplayMessagesSuccess)
+      receiverProbe.expectMsg(RecoverySuccess(5))
     }
     "replay a single message if count limit equals 1" in {
       journal ! ReplayMessages(2, 4, 1, pid, receiverProbe.ref)
       2 to 2 foreach { i => receiverProbe.expectMsg(replayedMessage(i)) }
-      receiverProbe.expectMsg(ReplayMessagesSuccess)
+      receiverProbe.expectMsg(RecoverySuccess(5))
     }
     "not replay messages if count limit equals 0" in {
       journal ! ReplayMessages(2, 4, 0, pid, receiverProbe.ref)
-      receiverProbe.expectMsg(ReplayMessagesSuccess)
+      receiverProbe.expectMsg(RecoverySuccess(5))
     }
     "not replay messages if lower  sequence number bound is greater than upper sequence number bound" in {
       journal ! ReplayMessages(3, 2, Long.MaxValue, pid, receiverProbe.ref)
-      receiverProbe.expectMsg(ReplayMessagesSuccess)
+      receiverProbe.expectMsg(RecoverySuccess(5))
     }
     "not replay permanently deleted messages (range deletion)" in {
-      val cmd = DeleteMessagesTo(pid, 3, true)
+      val cmd = DeleteMessagesTo(pid, 3, receiverProbe.ref)
       val sub = TestProbe()
 
       subscribe[DeleteMessagesTo](sub.ref)
       journal ! cmd
       sub.expectMsg(cmd)
 
+      receiverProbe.expectMsg(DeleteMessagesSuccess(3))
       journal ! ReplayMessages(1, Long.MaxValue, Long.MaxValue, pid, receiverProbe.ref)
       List(4, 5) foreach { i => receiverProbe.expectMsg(replayedMessage(i)) }
     }
+
+    // TODO: The below is not longer supported
```

Comment:
AFAIK logical deletes are no longer supported and ReadHighestSequenceNr is no longer in the public api.